api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -332,8 +332,13 @@ message ExternalProcessor {
 
   // Decorator to introduce custom logic that runs after a message received from
```

Comment:
Let's be clear here about the stage.  
I think we are implementing it as "run after the response is received, consumed and right before continue the filter chain". 

OTOH, if we change the contract here to that we run the "extension" right after the response is received, but before filter consuming it, would that simplify the code change needed quite a bit?

Reply:
In the general case, it's better to run the decorator method after the processing is done, so we can check the processing status to avoid wasted work. It might not always be possible for the extension to deduce the processing status.

Reply:
+1 to clearly document where is the response is passed to the extension, and after the processing makes sense to me.

Reply:
Modified comment to state that the decorator is invoked before continuing filter chain iteration.